synapse/storage/controllers/stats.py

```diff
@@ -13,8 +13,7 @@
 # limitations under the License.
 
 import logging
-from collections import Counter
-from typing import TYPE_CHECKING, Collection, List, Tuple
+from typing import TYPE_CHECKING, Collection, Counter, List, Tuple
```

Comment:
I'm somewhat surprised this works and that you don't need to separately import `Counter` from `collections` and `typing`!

Reply:
I don't think we need to worry about it:

```pycon
>>> from typing import Counter as CounterType
>>> from collections import Counter
>>> x = CounterType()
>>> isinstance(x, Counter)
True
```

Just a weird way to do it, I guess this is why they're making them scriptable.